### PR TITLE
Makes the iis_pool resource why run able

### PR DESCRIPTION
**Pull request by thomas-maurice**
* Fix the iis_pool resource to make it why run aware
* cleanup some obsolete parts of the code

### CODE REVIEW

**Comment by FlorentFlament** [comment]
:+1:

File: providers/pool.rb [hunk 1]
```diff
@@ -63,35 +69,41 @@
 
 action :start do
   if !@current_resource.running
-    shell_out!("#{appcmd(node)} start apppool \"#{site_identifier}\"")
-    new_resource.updated_by_last_action(true)
-    Chef::Log.info("#{new_resource} started")
+      execute "Starting IIS pool #{site_identifier}" do
```
Comment (guilhem) [inline providers/pool.rb]: Indentation?

File: providers/pool.rb [hunk 2]
```diff
@@ -280,16 +296,20 @@ def configure
       cmd << " \"/[name='#{new_resource.pool_name}'].processModel.identityType:SpecificUser\""
       cmd << " \"/[name='#{new_resource.pool_name}'].processModel.userName:#{new_resource.pool_username}\""
       cmd << " \"/[name='#{new_resource.pool_name}'].processModel.password:#{new_resource.pool_password}\"" if new_resource.pool_password && new_resource.pool_password != '' && is_new_password
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
+      converge_by("#{cmd}") do
+        Chef::Log.debug(cmd)
+        shell_out!(cmd)
```
Comment (guilhem) [inline providers/pool.rb]: Why not this one ?

File: providers/pool.rb [hunk 3]
```diff
@@ -38,23 +42,25 @@
     cmd << " /managedPipelineMode:#{new_resource.pipeline_mode.capitalize}" if new_resource.pipeline_mode
     cmd << " /commit:\"MACHINE/WEBROOT/APPHOST\""
     Chef::Log.debug(cmd)
```
Comment (guilhem) [inline providers/pool.rb]: `Chef::log` isn't so necessary as it is included by chef `execute` resource

**Comment by thomas-maurice** [comment]
fixing

**Review by guilhem — APPROVED** [review]
GG noRe

**Comment by pierrecdn** [comment]
LGTM :+1: , better would be to test that before merging into the `criteo` branch.

**Comment by pierrecdn** [comment]
Do not forget to propose those changes upstream via a PR, using one feature branch with all changes rebased on upstream/master, sign the CLA etc (https://docs.chef.io/community_contributions.html).

**Comment by thomas-maurice** [comment]
@pierrecdn da, but that would only make sense if I patched the other resources as well

**Pull request merged**